src/main/java/org/prebid/server/bidder/compass/CompassBidder.java

```diff
@@ -0,0 +1,173 @@
+                .uri(endpointUrl)
+                .headers(HttpUtil.headers())
+                .body(mapper.encodeToBytes(outgoingRequest))
+                .payload(outgoingRequest)
+                .build();
+    }
+
+    private Imp modifyImp(Imp imp, ExtImpCompass extImpCompass) {
+        try {
+            return imp.toBuilder()
+                    .ext(mapper.mapper().valueToTree(RequestBodyExt.of(getImpExtCompassWithType(extImpCompass))))
+                    .build();
+        } catch (IllegalArgumentException e) {
+            throw new PreBidException(e.getMessage());
+        }
+    }
+
+    private ImpExtCompass getImpExtCompassWithType(ExtImpCompass extImpCompass) {
+        ImpExtCompass impExtCompass = null;
+        if (!Objects.equals(extImpCompass.getEndpointId(), "")) {
+            impExtCompass = ImpExtCompass.builder()
+                    .type("network")
+                    .endpointId(extImpCompass.getEndpointId())
+                    .build();
+        } else if (!Objects.equals(extImpCompass.getPlacementId(), "")) {
+            impExtCompass = ImpExtCompass.builder()
+                    .type("publisher")
+                    .placementId(extImpCompass.getPlacementId())
+                    .build();
+        }
+        return impExtCompass;
+    }
+
+    private ExtImpCompass parseImpExt(Imp imp) {
+        try {
+            return mapper.mapper().convertValue(imp.getExt(), COMPASS_EXT_TYPE_REFERENCE).getBidder();
+        } catch (IllegalArgumentException e) {
+            throw new PreBidException(e.getMessage());
+        }
+    }
+
+    @Override
+    public Result<List<BidderBid>> makeBids(HttpCall<BidRequest> httpCall, BidRequest bidRequest) {
+        try {
+            final List<BidderError> errors = new ArrayList<>();
+            final BidResponse bidResponse = mapper.decodeValue(httpCall.getResponse().getBody(), BidResponse.class);
+            return Result.of(extractBids(httpCall.getRequest().getPayload(), bidResponse, errors), errors);
+        } catch (DecodeException | PreBidException e) {
+            return Result.withError(BidderError.badServerResponse(e.getMessage()));
+        }
+    }
+
+    private static List<BidderBid> extractBids(BidRequest bidRequest, BidResponse bidResponse,
+                                               List<BidderError> errors) {
+        if (bidResponse == null || CollectionUtils.isEmpty(bidResponse.getSeatbid())) {
+            return Collections.emptyList();
+        }
+        return bidsFromResponse(bidRequest, bidResponse, errors);
+    }
+
+    private static List<BidderBid> bidsFromResponse(BidRequest bidRequest, BidResponse bidResponse,
+                                                    List<BidderError> errors) {
+        return bidResponse.getSeatbid().stream()
+                .filter(Objects::nonNull)
+                .map(SeatBid::getBid)
+                .filter(Objects::nonNull)
+                .flatMap(Collection::stream)
```

Comment:
Add `filter(Objects::nonNull)` after this

Reply:
We can caught NPE on this `getBidType(bid.getImpid(), ...)`

Reply:
line 141 , we have `filter(Objects::nonNull)`